modules/InboundEmail/InboundEmail.php

```diff
@@ -5268,39 +5268,42 @@ public function returnNonImportedEmail($msgNo, $uid)
-
-                $dateTimeFormat = 'D, d M Y H:i:s O *';
-                if ($timezoneTextFound === false || $timezoneTextFound === 0) {
-                    $dateTimeFormat = 'D, d M Y H:i:s O';
+            if(empty($email->date_entered)) {
+
+                $possibleFormats = [
+                    \DateTime::RFC2822,
+                    str_replace(['D, '], '', \DateTime::RFC2822), // day-of-week is optional
+                    str_replace([':s'], '', \DateTime::RFC2822), // seconds are optional
+                    str_replace(['D, ', ':s'], '', \DateTime::RFC2822), // day-of-week is optional, seconds are optional
+                    \DateTime::RFC822,
+                    str_replace(['D, '], '', \DateTime::RFC822), // day is optional
+                    str_replace([':s'], '', \DateTime::RFC822), // seconds are optional
+                    str_replace(['D, ', ':s'], '', \DateTime::RFC822), // day is optional, seconds are optional
+                ];
+
+                foreach ($possibleFormats  as $possibleFormat) {
+                    $dateTime = \DateTime::createFromFormat( $possibleFormat, $header[0]->date);
+                    if ($dateTime !== false) {
+                        break;
+                    }
                 }
 
-                $dateTime = DateTime::createFromFormat(
-                    $dateTimeFormat,
-                    $header[0]->date
-                );
-
-                if (!empty($dateTime)) {
-                    $email->date_entered = $timedate->asUser($dateTime, $current_user);
-                    $email->date_modified = $timedate->asUser($dateTime, $current_user);
-                    $email->date_start = $timedate->asUserDate($dateTime);
-                    $email->time_start = $timedate->asUserTime($dateTime);
-
-                    $systemUser = BeanFactory::getBean('Users', 1);
-                    $email->created_by = $systemUser->id;
-                    $email->created_by_name = $systemUser->name;
-                    $email->modified_user_id = $systemUser->id;
-                    $email->modified_by_name = $systemUser->name;
-                } else {
+                if ($dateTime === false) {
                     throw new Exception(
-                        'DateTime::createFromFormat (' . $dateTimeFormat . ',' . $header[0]->date . '): ' .
-                        'expected DateTime but it returned FALSE or empty.'
+                        sprintf('Expected header Date to comply with RFC882 or RFC2882, but actual is "%s"', $header[0]->date)
                     );
                 }
+
+                $email->date_entered = $timedate->asUser($dateTime, $current_user);
+                $email->date_modified = $timedate->asUser($dateTime, $current_user);
+                $email->date_start = $timedate->asUserDate($dateTime);
+                $email->time_start = $timedate->asUserTime($dateTime);
+
+                $systemUser =  BeanFactory::getBean('Users', 1);
```

Comment:
if getting the system user is really necessary, it would be better to use the user function getSystemUser, as it not guaranteed the user with id 1 exists

Reply:
I don't know enough about that to change it.  I would argue it shouldn't be a part of this PR.